% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={ImprovAFish\_high\_dosage},
  pdfauthor={Shashank Gupta},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{ImprovAFish\_high\_dosage}
\author{Shashank Gupta}
\date{2023-02-03}

\begin{document}
\maketitle

\hypertarget{feed-microbiome-host-interactions-in-atlantic-salmon-over-life-stages}{%
\section{Feed-microbiome-host interactions in Atlantic salmon over life
stages}\label{feed-microbiome-host-interactions-in-atlantic-salmon-over-life-stages}}

To meet future food demands, more efficient and sustainable animal
production systems are needed. Given the crucial importance of the gut
microbiota to animal (host) health and nutrition, selective enhancement
of beneficial microbes via prebiotics may be a powerful approach for
promoting farmed fish welfare and robustness. In this study, we employed
three versions of a beta-mannan prebiotic that were fed to Atlantic
salmon and explored the combined responses of both gut microbiota and
the host from freshwater to seawater life stages. We have used weighted
gene co-expression network analysis (WGCNA) of host RNA-seq and
microbial 16S rRNA amplicon sequencing data to identify biological
interactions between the gut ecosystem and the host. We observed several
microbial and host modules through WGCNA that were significantly
correlated with life stage, but not with diet. Microbial diversity was
highest in the early life of Atlantic salmon and decreased over time. In
particular, Lactobacillus and Paraburkholderia were the dominating
genera and showed the highest correlation with host modules. Our
findings demonstrate that salmon-microbiota interactions are mainly
influenced by life stage, while further research is required to
determine whether supplementation of selected prebiotics to diet can be
used to modulate the salmon gut microbiota for improving host health and
production sustainability.

\hypertarget{getting-started}{%
\subsection{Getting Started}\label{getting-started}}

\hypertarget{step-1.-package-dependencies-in-r}{%
\subsubsection{Step 1. Package dependencies in
R}\label{step-1.-package-dependencies-in-r}}

Installing R packages can be done through various sources such as
GitHub, the Comprehensive R Archive Network (CRAN), or by following the
official website of the package.

To install a package from GitHub, use the devtools package and the
\texttt{install\_github()} function. For example, to install the
``ggplot2'' package from GitHub, run the following code:

\begin{verbatim}
library(devtools)
install_github("ggplot2")
\end{verbatim}

To install a package from CRAN, use the \texttt{install.packages()}
function. For example, to install the ``dplyr'' package from CRAN, run
the following code:

\begin{verbatim}
install.packages("dplyr")
\end{verbatim}

Finally, to install a package from its official website, download the
package source code, and use the \texttt{install.packages()} function
with the local file path as the argument. For example, to install the
``reshape2'' package from its official website, first download the
source code, then run the following code:

\begin{verbatim}
install.packages("path/to/reshape2_package.tar.gz", repos = NULL, type = "source")
\end{verbatim}

It is recommended to regularly update the installed packages to ensure
compatibility and to benefit from new features and bug fixes.

\hypertarget{step-2.-shell-script}{%
\subsubsection{Step 2. Shell script}\label{step-2.-shell-script}}

Primers were removed from the raw paired-end FASTQ files generated via
MiSeq using ``cutadapt''. Further, reads were analyzed by QIIME2
(qiime2-2021.8) pipeline through dada2 to infer the ASVs present and
their relative abundances across the samples. For bed dust samples,
using read quality scores for the dataset, forward and reverse reads
were truncated at 280 bp and 260 bp, followed by trimming the 5′ end
till 25 bp for both forward and reverse reads, respectively; other
quality parameters used dada2 default values for both 16S rRNA gene
sequencing. For 16S rRNA gene sequencing, taxonomy was assigned using a
pre-trained Naïve Bayes classifier (Silva database, release 138, 99\%
ASV) were used.

The code is a shell script for processing paired-end sequencing data in
order to perform a microbial analysis. It uses a combination of bash
commands and QIIME2 (Quantitative Insights Into Microbial Ecology)
commands.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ls }\SpecialCharTok{{-}}\NormalTok{d }\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{ data}\SpecialCharTok{/}\ErrorTok{*}\NormalTok{\_R1}\SpecialCharTok{*} \ErrorTok{\textgreater{}}\NormalTok{ forward.txt}
\NormalTok{ls }\SpecialCharTok{{-}}\NormalTok{d }\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{ data}\SpecialCharTok{/}\ErrorTok{*}\NormalTok{\_R2}\SpecialCharTok{*} \ErrorTok{\textgreater{}}\NormalTok{ reverse.txt}

\NormalTok{mkdir Trimmed}
\NormalTok{mkdir Trim.log}

\NormalTok{parallel }\SpecialCharTok{{-}}\NormalTok{j }\DecValTok{1} \SpecialCharTok{{-}{-}}\NormalTok{xapply }\StringTok{"cutadapt {-}g CCTAYGGGRBGCASCAG {-}G GGACTACHVGGGTWTCTAAT {-}{-}pair{-}filter=any {-}{-}discard{-}untrimmed {-}o $PWD/Trimmed/\{1/.\}.gz {-}p $PWD/Trimmed/\{2/.\}.gz \{1\} \{2\} \&\textgreater{} $PWD/Trim.log/\{1/.\}.log"} \SpecialCharTok{:::}\ErrorTok{:}\NormalTok{ forward.txt }\SpecialCharTok{:::}\ErrorTok{:}\NormalTok{ reverse.txt}

\NormalTok{qiime tools import }\SpecialCharTok{{-}{-}}\NormalTok{type }\StringTok{\textquotesingle{}SampleData[PairedEndSequencesWithQuality]\textquotesingle{}} \SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{format CasavaOneEightSingleLanePerSampleDirFmt }\SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path Trimmed}\SpecialCharTok{/} \SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path demux}\SpecialCharTok{{-}}\NormalTok{paired}\SpecialCharTok{{-}}\NormalTok{end.qza}

\NormalTok{qiime demux summarize }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{data demux}\SpecialCharTok{{-}}\NormalTok{paired}\SpecialCharTok{{-}}\NormalTok{end.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{visualization step1\_output.qzv}

\NormalTok{qiime dada2 denoise}\SpecialCharTok{{-}}\NormalTok{paired }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{demultiplexed}\SpecialCharTok{{-}}\NormalTok{seqs demux}\SpecialCharTok{{-}}\NormalTok{paired}\SpecialCharTok{{-}}\NormalTok{end.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{table table }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{representative}\SpecialCharTok{{-}}\NormalTok{sequences rep}\SpecialCharTok{{-}}\NormalTok{seqs  }\SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{trunc}\SpecialCharTok{{-}}\NormalTok{len}\SpecialCharTok{{-}}\NormalTok{f }\DecValTok{270} \SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{trunc}\SpecialCharTok{{-}}\NormalTok{len}\SpecialCharTok{{-}}\NormalTok{r }\DecValTok{250} \SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{n}\SpecialCharTok{{-}}\NormalTok{threads }\DecValTok{4} \SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{denoising}\SpecialCharTok{{-}}\NormalTok{stats denoising}\SpecialCharTok{{-}}\NormalTok{stats.qza }\SpecialCharTok{{-}{-}}\NormalTok{verbose }\SpecialCharTok{\&}\ErrorTok{\textgreater{}}\NormalTok{ dada2.log}

\NormalTok{qiime metadata tabulate }\SpecialCharTok{{-}{-}}\NormalTok{m}\SpecialCharTok{{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{file denoising}\SpecialCharTok{{-}}\NormalTok{stats.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{visualization step2\_output.qzv}

\CommentTok{\# Creating Rooted Phylogenetic Tree}
\NormalTok{echo }\StringTok{"Align rep{-}seqs and created rooted tree"}
\NormalTok{qiime alignment mafft }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{sequences rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{alignment aligned}\SpecialCharTok{{-}}\NormalTok{rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{n}\SpecialCharTok{{-}}\NormalTok{threads }\DecValTok{4}
\NormalTok{qiime alignment mask }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{alignment aligned}\SpecialCharTok{{-}}\NormalTok{rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{masked}\SpecialCharTok{{-}}\NormalTok{alignment masked}\SpecialCharTok{{-}}\NormalTok{aligned}\SpecialCharTok{{-}}\NormalTok{rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza}
\NormalTok{qiime phylogeny fasttree }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{alignment masked}\SpecialCharTok{{-}}\NormalTok{aligned}\SpecialCharTok{{-}}\NormalTok{rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{tree unrooted}\SpecialCharTok{{-}}\NormalTok{tree.qza }\SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{n}\SpecialCharTok{{-}}\NormalTok{threads }\DecValTok{4}
\NormalTok{qiime phylogeny midpoint}\SpecialCharTok{{-}}\NormalTok{root }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{tree unrooted}\SpecialCharTok{{-}}\NormalTok{tree.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{rooted}\SpecialCharTok{{-}}\NormalTok{tree rooted}\SpecialCharTok{{-}}\NormalTok{tree.qza}
\NormalTok{echo }\StringTok{"Rooted tree has been created"}

\CommentTok{\# Taxonomy classification}
\NormalTok{echo }\StringTok{"Classify rep{-}seqs using Silva132\_99 classifier"}
\NormalTok{qiime feature}\SpecialCharTok{{-}}\NormalTok{classifier classify}\SpecialCharTok{{-}}\NormalTok{sklearn }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{reads rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{classifier silva\_99\_341F806R\_classifier.qza }\SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{n}\SpecialCharTok{{-}}\NormalTok{jobs }\DecValTok{6} \SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{classification taxonomy.qza}
\NormalTok{echo }\StringTok{"Classification done"}

\CommentTok{\# Export to biom file and import into phyloseq}
\NormalTok{echo }\StringTok{"Export data from qiime"}
\NormalTok{qiime tools export  }\SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path table.qza }\SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}
\NormalTok{qiime tools export  }\SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path taxonomy.qza }\SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}
\NormalTok{echo }\StringTok{"data has been exported"}
\CommentTok{\# change header of taxonomy file}
\NormalTok{sed }\SpecialCharTok{{-}}\NormalTok{i }\StringTok{"1 s/\^{}.*$/\#OTUID}\SpecialCharTok{\textbackslash{}t}\StringTok{taxonomy}\SpecialCharTok{\textbackslash{}t}\StringTok{confidence/"}\NormalTok{ exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}\SpecialCharTok{/}\NormalTok{taxonomy.tsv}
\CommentTok{\# fuse}
\NormalTok{biom add}\SpecialCharTok{{-}}\NormalTok{metadata }\SpecialCharTok{{-}}\NormalTok{i exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}\SpecialCharTok{/}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table.biom }\SpecialCharTok{{-}}\NormalTok{o exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}\SpecialCharTok{/}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table\_taxonomy.biom }\SpecialCharTok{{-}{-}}\NormalTok{observation}\SpecialCharTok{{-}}\NormalTok{metadata}\SpecialCharTok{{-}}\NormalTok{fp exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}\SpecialCharTok{/}\NormalTok{taxonomy.tsv }\SpecialCharTok{{-}{-}}\NormalTok{observation}\SpecialCharTok{{-}}\NormalTok{header OTUID,taxonomy,confidence }\SpecialCharTok{{-}{-}}\NormalTok{sc}\SpecialCharTok{{-}}\NormalTok{separated taxonomy}
\NormalTok{echo }\StringTok{"BIOM file with OTU table and taxonomy saved as: exported{-}feature{-}table/feature{-}table\_taxonomy.biom"}
\CommentTok{\# export tree}
\NormalTok{qiime tools export }\SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path  rooted}\SpecialCharTok{{-}}\NormalTok{tree.qza }\SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}\SpecialCharTok{/}
\NormalTok{echo }\StringTok{"Rooted phylogenetic tree saved as: exported{-}feature{-}table/tree.nwk"}
\CommentTok{\# export rep{-}seqs}
\NormalTok{qiime tools export  }\SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path merged\_rep}\SpecialCharTok{{-}}\NormalTok{seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path exported}\SpecialCharTok{{-}}\NormalTok{feature}\SpecialCharTok{{-}}\NormalTok{table}\SpecialCharTok{/}
\NormalTok{echo }\StringTok{"Representative sequences saved as: exported{-}feature{-}table/dna{-}sequences.fasta"}

\CommentTok{\# Training the database}
\NormalTok{wget https}\SpecialCharTok{:}\ErrorTok{//}\NormalTok{www.arb}\SpecialCharTok{{-}}\NormalTok{silva.de}\SpecialCharTok{/}\NormalTok{fileadmin}\SpecialCharTok{/}\NormalTok{silva\_databases}\SpecialCharTok{/}\NormalTok{qiime}\SpecialCharTok{/}\NormalTok{Silva\_132\_release.zip}
\NormalTok{unzip Silva\_132\_release.zip}

\NormalTok{qiime tools import }\SpecialCharTok{{-}{-}}\NormalTok{type }\StringTok{\textquotesingle{}FeatureData[Sequence]\textquotesingle{}} \SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path .}\SpecialCharTok{/}\NormalTok{SILVA\_132\_QIIME\_release}\SpecialCharTok{/}\NormalTok{rep\_set}\SpecialCharTok{/}\NormalTok{rep\_set\_16S\_only}\SpecialCharTok{/}\DecValTok{99}\SpecialCharTok{/}\NormalTok{silva\_132\_99\_16S.fna  }\SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path silva\_99\_seqs.qza}

\NormalTok{qiime tools import }\SpecialCharTok{{-}{-}}\NormalTok{type }\StringTok{\textquotesingle{}FeatureData[Taxonomy]\textquotesingle{}} \SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{format HeaderlessTSVTaxonomyFormat }\SpecialCharTok{{-}{-}}\NormalTok{input}\SpecialCharTok{{-}}\NormalTok{path .}\SpecialCharTok{/}\NormalTok{SILVA\_132\_QIIME\_release}\SpecialCharTok{/}\NormalTok{taxonomy}\SpecialCharTok{/}\NormalTok{16S\_only}\SpecialCharTok{/}\DecValTok{99}\SpecialCharTok{/}\NormalTok{consensus\_taxonomy\_all\_levels.txt }\SpecialCharTok{{-}{-}}\NormalTok{output}\SpecialCharTok{{-}}\NormalTok{path silva\_99\_consensus\_taxonomy.qza}

\NormalTok{qiime feature}\SpecialCharTok{{-}}\NormalTok{classifier extract}\SpecialCharTok{{-}}\NormalTok{reads }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{sequences silva\_99\_seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{f}\SpecialCharTok{{-}}\NormalTok{primer CCTAYGGGRBGCASCAG }\SpecialCharTok{{-}{-}}\NormalTok{p}\SpecialCharTok{{-}}\NormalTok{r}\SpecialCharTok{{-}}\NormalTok{primer GGACTACHVGGGTWTCTAAT }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{reads silva\_99\_341F806R.seqs.qza}


\NormalTok{qiime feature}\SpecialCharTok{{-}}\NormalTok{classifier fit}\SpecialCharTok{{-}}\NormalTok{classifier}\SpecialCharTok{{-}}\NormalTok{naive}\SpecialCharTok{{-}}\NormalTok{bayes }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{reference}\SpecialCharTok{{-}}\NormalTok{reads silva\_99\_341F806R.seqs.qza }\SpecialCharTok{{-}{-}}\NormalTok{i}\SpecialCharTok{{-}}\NormalTok{reference}\SpecialCharTok{{-}}\NormalTok{taxonomy silva\_99\_consensus\_taxonomy.qza }\SpecialCharTok{{-}{-}}\NormalTok{o}\SpecialCharTok{{-}}\NormalTok{classifier silva\_99\_341F806R\_classifier.qza }\SpecialCharTok{{-}{-}}\NormalTok{verbose }\SpecialCharTok{\&}\ErrorTok{\textgreater{}}\NormalTok{ silva\_99\_341F806R\_classifier.log}
\end{Highlighting}
\end{Shaded}

All downstream analyses were performed on this normalized ASVs table
unless mentioned. We used two alpha diversity indices, i.e., observed
richness and Shannon diversity index. Furthermore, beta diversity was
calculated using weighted and unweighted UniFrac metric and visualized
by principal coordinates analysis (PCoA). Alpha and beta diversity was
calculated using phyloseq v1.38.0 and visualized with ggplot2 v3.3.5 in
R v4.1.1. Comparison of community richness and diversity was assessed by
the Kruskal-Wallis test between all the groups, and comparison between
the two groups was done by Wilcoxon test with Benjamini-Hochberg FDR
multiple test correction. Significance testing between the groups for
beta diversity was assessed using permutational multivariate analysis of
variance (PERMANOVA) using the ``vegan'' package.

\hypertarget{step-3.-downstream-analysis-with-r}{%
\subsubsection{Step 3. Downstream analysis with
R}\label{step-3.-downstream-analysis-with-r}}

We require several packages-

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(}\StringTok{"ranacapa"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"phyloseq"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"ggplot2"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"stringr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"plyr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"reshape2"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"reshape"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"dplyr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"tidyr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"doBy"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"plyr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"microbiome"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"ggpubr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"vegan"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"tidyverse"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"magrittr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"cowplot"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"dendextend"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"WGCNA"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"metagenomeSeq"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"decontam"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"RColorBrewer"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(}\StringTok{"ampvis2"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}


\end{document}
